src/Base/MatrixPyImp.cpp

```diff
@@ -301,27 +308,39 @@ PyObject* MatrixPy::scale(PyObject * args)
     Base::Vector3d vec;
     PyObject *pcVecObj;
 
-    if (PyArg_ParseTuple(args, "ddd", &x,&y,&z)) {
-        vec.x = x;
-        vec.y = y;
-        vec.z = z;
-    }
-    else if (PyArg_ParseTuple(args, "O!:three floats or a vector is needed",
-        &PyTuple_Type, &pcVecObj)) {
-        vec = getVectorFromTuple<double>(pcVecObj);
-        // clears the error from the first PyArg_ParseTuple()6
+    do { // dummy do..while for cascaded if
+        if (PyArg_ParseTuple(args, "ddd", &x,&y,&z)) {
+            vec.x = x;
+            vec.y = y;
+            vec.z = z;
+            break;
+        }
+        // clears the error from previous PyArg_ParseTuple()
         PyErr_Clear();
-    }
-    else if (PyArg_ParseTuple(args, "O!:three floats or a vector is needed", &(Base::VectorPy::Type), &pcVecObj)) {
-        // convert args: Python->C
-        Base::VectorPy  *pcObject = static_cast<Base::VectorPy*>(pcVecObj);
-        Base::Vector3d* val = pcObject->getVectorPtr();
-        vec.Set(val->x,val->y,val->z);
-        // clears the error from the first PyArg_ParseTuple()6
+        if (PyArg_ParseTuple(args, "d", &x)) {
+            vec.x = vec.y = vec.z = x;
+            break;
+        }
+        // clears the error from previous PyArg_ParseTuple()
+        PyErr_Clear();
+        if (PyArg_ParseTuple(args, "O!:three floats or a vector is needed",
+            &PyTuple_Type, &pcVecObj)) {
+            vec = getVectorFromTuple<double>(pcVecObj);
+            break;
+        }
+        // clears the error from previous PyArg_ParseTuple()
+        PyErr_Clear();
+        if (PyArg_ParseTuple(args, "O!:three floats or a vector is needed", &(Base::VectorPy::Type), &pcVecObj)) {
+            Base::VectorPy  *pcObject = static_cast<Base::VectorPy*>(pcVecObj);
+            Base::Vector3d* val = pcObject->getVectorPtr();
+            vec.Set(val->x,val->y,val->z);
+            break;
+        }
+        // clears the error from previous PyArg_ParseTuple()
         PyErr_Clear();

```

Comment:
Returning `nullptr` without setting an exception raises a `SystemError` exception.
This is the case if the arguments passed to the function do not meet the requirements in any of the previous `PyArg_ParseTuple` calls.
So, don't call `PyErr_Clear()` here, or instead set a specific error with `PyErr_SetString` before returning `nullptr`:

`PyErr_SetString(PyExc_TypeError, "Expected three floats or tuple with three floats or Base.Vector");`
`return nullptr;`

If you decide to use this approach, remove the error message `"three floats or a vector is needed"` from the format string in `PyArg_ParseTuple`,  which should actually be set using a semicolon, not a colon.

Reply:
Good catch. Thanks for fast review.